### PR TITLE
[Zen2] Update default for USE_ZEN2 to true

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today the default for USE_ZEN2 is false and it is overridden in many places. By
defaulting it to true we can be sure that the only places in which Zen2 does
not work are those in which it is explicitly set to false.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/client/transport/TransportClientIT.java [hunk 1]
```diff
@@ -66,7 +66,6 @@ public void testNodeVersionIsUpdated() throws IOException, NodeValidationExcepti
                 .put("transport.type", getTestTransportType())
                 .put(Node.NODE_DATA_SETTING.getKey(), false)
                 .put("cluster.name", "foobar")
-                .put(TestZenDiscovery.USE_ZEN2.getKey(), true)
```
Comment (ywelsch) [inline server/src/test/java/org/elasticsearch/client/transport/TransportClientIT.java]: This (and some of the other occurrences removed in this PR) should be set to Zen1 or Zen2 depending on whether the surrounding test uses the corresponding disco type. If we want to randomize the tests to run with Zen1 / Zen2 for a while, it's probably better to add a field `useZen2` to ESTestCase (similar to `useNio`) that will for now be set to Zen2, but could also be set to randomBoolean() for a while. This line here should then be changed to `.put(TestZenDiscovery.USE_ZEN2.getKey(), getUseZen2())`

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java [hunk 2]
```diff
@@ -1925,8 +1925,8 @@ protected NodeConfigurationSource getNodeConfigSource() {
             initialNodeSettings.put(NetworkModule.TRANSPORT_TYPE_KEY, getTestTransportType());
             initialTransportClientSettings.put(NetworkModule.TRANSPORT_TYPE_KEY, getTestTransportType());
         }
-        if (addTestZenDiscovery()) {
-            initialNodeSettings.put(TestZenDiscovery.USE_ZEN2.getKey(), true);
+        if (addTestZenDiscovery() == false) {
+            initialNodeSettings.put(TestZenDiscovery.USE_ZEN2.getKey(), false);
```
Comment (ywelsch) [inline test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: this condition is non-sensical now. Only TestZenDiscovery understands the `TestZenDiscovery.USE_ZEN2` setting

Comment (DaveCTurner) [reply test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: Yes, fixed in ace9194.

File: server/src/test/java/org/elasticsearch/cluster/coordination/PersistedStateIT.java [hunk 3]
```diff
@@ -33,7 +33,7 @@
 public class PersistedStateIT extends ESIntegTestCase {
 
     private static Settings SETTINGS = Settings.builder()
-        .put(TestZenDiscovery.USE_ZEN2.getKey(), true)
+        .put(TestZenDiscovery.USE_ZEN2.getKey(), getUseZen2())
```
Comment (ywelsch) [inline server/src/test/java/org/elasticsearch/cluster/coordination/PersistedStateIT.java]: this test was meant to specifically test Zen2
so maybe just set it to true here.
Note that this test can probably go away once state recovery is implemented.

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
Last change was trivial and passed precommit, so merging.

**Pull request merged**